File: back/app/src/main/java/com/schedule/app/dto/RegularScheduleDTO.java
package com.schedule.app.dto;

import java.time.DayOfWeek;
import java.time.LocalDate;
import java.time.LocalTime;
import lombok.Builder;

@Builder
public record RegularScheduleDTO(
    int scheduleId,
    LocalTime startTime,
    LocalTime endTime,
    LocalDate startDate,
    LocalDate endDate,
    DayOfWeek daysOfWeek,
    String worktypeName,
    String worktypeColor) {}

File: front/src/type/dto/regularschedule.dto.ts
import { DayOfWeek } from '@/enum/dayofweek.enum';

/**
 * レギュラスケジュールDTO
 */
export type RegularscheduleDTO = {
  scheduleId: string;
  daysOfWeek: DayOfWeek;
  startTime: string;
  endTime: string;
  startDate: string;
  endDate: string;
  worktypeName: string;
  worktypeColor: string;
};
